components/push/src/communications.rs

```diff
@@ -274,7 +274,12 @@ impl Connection for ConnectHttp {
             .headers(self.headers()?)
             .send()
         {
-            Ok(_) => Ok(true),
+            Ok(_) => {
+                if channel_id.is_none() {
+                    self.uaid = None;
+                }
```

Comment:
Not for this PR, but something to link about more architecturally for this component - if the `DELETE` request succeeds but then we fail to persist this `self.uaid` change to disk due to e.g. a crash or unrelated error, then we'll still be in the same position of disagreement between client and server and will need to be prepared to deal with it.

This is definitely still a big improvement over the previous behaviour!